code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushNotificationBuilder.java

```diff
@@ -0,0 +1,502 @@
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile;
+
+import static com.adobe.marketing.mobile.CampaignPushConstants.LOG_TAG;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.TaskStackBuilder;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.media.RingtoneManager;
+import android.os.Build;
+import android.widget.RemoteViews;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.StringUtils;
+import com.adobe.marketing.mobile.campaignclassic.R.layout;
+import com.adobe.marketing.mobile.campaignclassic.R.id;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+
+/**
+ * Class for building push notifications.
+ * <p>
+ * The {@link #buildPushNotification(AEPPushPayload, Context)} method in this class takes the {@link AEPPushPayload} created from the
+ * push notification and builds the notification.
+ * This class is used internally by the {@link AEPMessagingService} to build the push notification.
+ */
+class AEPPushNotificationBuilder {
+
+    private static final String SELF_TAG = "AEPPushNotificationBuilder";
+    private static final String DEFAULT_CHANNEL_ID = "CampaignPushChannel";
+    // When no channel name is received from the push notification, this default channel name is used.
+    // This will appear in the notification settings for the app.
+    private static final String DEFAULT_CHANNEL_NAME = "Campaign Classic General Notifications";
+
+    /**
+     * Builds a notification for the provided {@code AEPPushPayload}.
+     *
+     * @param payload {@link AEPPushPayload} created from the received push notification
+     * @param context the application {@link Context}
+     * @return the notification
+     */
+    @Nullable
+    static Notification buildPushNotification(final AEPPushPayload payload,
+                                              final Context context) {
+        Notification notification;
+        final Map<String, String> messageData = payload.getMessageData();
+        final PushTemplateType pushTemplateType = PushTemplateType.fromString((Objects.requireNonNull(messageData.get(CampaignPushConstants.PushPayloadKeys.TEMPLATE_TYPE))));
```

Comment:
`Objects.requireNonNull` with throw an `NullPointerException` anyway in case of an error scenario. 
Given that `PushTemplateType.fromString` returns `UNKNOWN` when template type cannot be resolved, do you still require this check?

Reply:
this is no longer required and i will clean it up